okhttp-tls/src/main/kotlin/okhttp3/tls/internal/TlsUtil.kt

```diff
@@ -64,7 +67,13 @@ object TlsUtil {
       "Unexpected trust managers: ${result.contentToString()}"
     }
 
-    return result[0] as X509TrustManager
+    val trustManager = result[0] as X509TrustManager
+
+    return when {
+      insecureHosts.isEmpty() -> trustManager
+      AndroidPlatform.isAndroid -> InsecureAndroidTrustManager(trustManager, insecureHosts)
+      else -> InsecureExtendedTrustManager(trustManager as X509ExtendedTrustManager, insecureHosts)
```

Comment:
Hooking in here is smart! Less to worry about with the lifecycle of reusing clients via newBuilder etc etc.

Reply:
Yeah. I originally wanted this in OkHttpClient.Builder but I couldn’t find a nice way to change the behavior of the user’s supplied trust manager.

Reply:
Good call.